packages/stock-location-next/src/services/stock-location.ts

```diff
@@ -134,14 +180,24 @@ export default class StockLocationModuleService<
    */
   @InjectManager("baseRepository_")
   async update(
+    idOrSelector: string | FilterableStockLocationProps,
     data: UpdateStockLocationNextInput | UpdateStockLocationNextInput[],
     @MedusaContext() context: Context = {}
   ): Promise<
     StockLocationTypes.StockLocationDTO | StockLocationTypes.StockLocationDTO[]
   > {
-    const input = Array.isArray(data) ? data : [data]
-
-    const updated = await this.update_(input, context)
+    let normalizedInput: (UpdateStockLocationNextInput & { id: string })[] = []
+    if (isString(idOrSelector)) {
+      normalizedInput = [{ id: idOrSelector, ...data }]
+    } else {
+      const locations = await this.list(idOrSelector, {}, context)
+
+      normalizedInput = locations.map((region) => ({
+        id: region.id,
+        ...data,
+      }))
+    }
+    const updated = await this.update_(normalizedInput, context)
```

Comment:
suggestion: if instead of calling update_ you call `this.stockLocationService_.update` you wouldn't have to do the if else before as the internal update accepts a selector too.

cc @olivermrbl